src/state/apis/zapper/zapperApi.ts

```diff
@@ -541,43 +566,35 @@ export const zapper = createApi({
-                        ? bnOrZero(
-                            bnOrZero(bnOrZero(token0ReservesCryptoPrecision.toFixed()).toString()),
-                          ).times(bn(10).pow(asset.decimals ?? 18))
-                        : undefined
-                    const token1ReservesBaseUnit =
-                      typeof token1ReservesCryptoPrecision === 'number'
-                        ? bnOrZero(
-                            bnOrZero(bnOrZero(token1ReservesCryptoPrecision.toFixed()).toString()),
-                          ).times(bn(10).pow(asset.decimals ?? 18))
-                        : undefined
-                    const totalSupplyBaseUnit =
-                      typeof asset.supply === 'number'
-                        ? bnOrZero(asset.supply)
-                            .times(bn(10).pow(asset.decimals ?? 18))
-                            .toString()
-                        : undefined
-                    const token0PoolRatio =
-                      token0ReservesBaseUnit && totalSupplyBaseUnit
-                        ? token0ReservesBaseUnit.div(totalSupplyBaseUnit).toString()
-                        : undefined
-                    const token1PoolRatio =
-                      token1ReservesBaseUnit && totalSupplyBaseUnit
-                        ? token1ReservesBaseUnit.div(totalSupplyBaseUnit).toString()
-                        : undefined
-
-                    if (!(token0PoolRatio && token1PoolRatio)) return
-
-                    return [
-                      toBaseUnit(token0PoolRatio.toString(), asset.tokens[0].decimals),
-                      toBaseUnit(token1PoolRatio.toString(), asset.tokens[1].decimals),
-                    ] as const
-                  })()
+                  const underlyingAssetRatiosBaseUnit = (() =>
+                    (asset.dataProps?.reserves ?? [])
+                      .map(reserve => {
+                        const reserveBaseUnit = bnOrZero(
+                          bnOrZero(bnOrZero(reserve).toFixed()).toString(),
+                        ).times(bn(10).pow(asset.decimals ?? 18))
+
+                        const totalSupplyBaseUnit =
+                          typeof asset.supply === 'number'
+                            ? bnOrZero(asset.supply)
+                                .times(bn(10).pow(asset.decimals ?? 18))
+                                .toString()
+                            : undefined
+
+                        const tokenPoolRatio =
+                          reserveBaseUnit && totalSupplyBaseUnit
+                            ? reserveBaseUnit.div(totalSupplyBaseUnit).toString()
+                            : undefined
+
+                        if (!tokenPoolRatio) return undefined
+                        const ratio = toBaseUnit(
+                          tokenPoolRatio.toString(),
+                          asset.tokens[0].decimals,
+                        )
+                        return ratio
```

Comment:
```suggestion
                      const reserveBaseUnit = toBaseUnit(reserve, asset.decimals ?? 18)
                      const totalSupplyBaseUnit =
                        typeof asset.supply === 'number'
                          ? toBaseUnit(asset.supply, asset.decimals ?? 18)
                          : undefined
                      const tokenPoolRatio = totalSupplyBaseUnit
                        ? bn(reserveBaseUnit).div(totalSupplyBaseUnit).toString()
                        : undefined
                      if (!tokenPoolRatio) return '0'
                      const ratio = toBaseUnit(tokenPoolRatio, asset.tokens[0].decimals)
                      return ratio
```

Reply:
this would require update to the `underlyingAssetRatiosBaseUnit` type to be string[]